src/NITTA/Model/Microarchitecture/Builder.hs

```diff
@@ -70,31 +78,38 @@ defineNetwork bnName ioSync builder =
             , bnPus = M.fromList puProtos
             , bnSignalBusWidth = signalBusWidth
             , ioSync
-            , bnEnv = defEnv{ioPorts = Just netIOPorts}
+            , bnEnv = def{ioPorts = Just $ netIOPorts puProtos}
+            , bnAvailPorts = availPorts
+            }
+
+modifyNetwork net@BusNetwork{bnPus, bnSignalBusWidth, bnAvailPorts, bnEnv} builder =
+    let st0 =
+            BuilderSt
+                { signalBusWidth = bnSignalBusWidth
+                , availPorts = bnAvailPorts
+                , puProtos = M.toList bnPus
+                }
+        BuilderSt{signalBusWidth, puProtos, availPorts} = flip execState st0 $ void builder
```

Comment:
I don’t feel that void here is needed. Because without it, you move to restrict builder type to return nothing, which decreases the chance of mistake of user.